app/controllers/laa_references_controller.rb

```diff
@@ -1,42 +1,49 @@
 # frozen_string_literal: true
 
 class LaaReferencesController < ApplicationController
-  include DefendantSearchable
-
-  before_action :set_defendant_asn_or_nino_if_required,
-                :load_and_authorize_search,
+  before_action :set_defendant_uuid_if_required,
+                :authorize_defendant_search,
                 :set_link_attempt,
                 :set_defendant_if_required
 
   add_breadcrumb :search_filter_breadcrumb_name, :new_search_filter_path
   add_breadcrumb :search_breadcrumb_name, :search_breadcrumb_path
 
-  add_breadcrumb (proc { |v| v.prosecution_case_name(v.controller.defendant.prosecution_case_reference) }),
-                 (proc { |v| v.prosecution_case_path(v.controller.defendant.prosecution_case_reference) })
+  # TODO: I have (temporarily) removed the case breadcrumb as this was
+  #       causing issues.
+  # add_breadcrumb (proc { |v| v.prosecution_case_name(v.controller.defendant.asn) }),
+  #                (proc { |v| v.prosecution_case_path(v.controller.defendant.id) })
 
   def new
     add_breadcrumb defendant.name,
-                   defendant_path(defendant.arrest_summons_number || defendant.national_insurance_number)
+                   defendant_path(defendant.id)
   end
 
   def create
     authorize! :create, :link_maat_reference, message: I18n.t('unauthorized.default')
 
     if @link_attempt.valid?
-      if link_laa_reference
-        flash[:notice] = I18n.t('laa_reference.link.success')
-        redirect_to edit_defendant_path(id: @defendant_asn_or_nino)
-      else
-        flash[:alert] = I18n.t('laa_reference.link.failure', error_messages: error_messages)
-        redirect_to new_laa_reference_path(id: @defendant_asn_or_nino)
-      end
+      redirect_after_link
     else
       render 'new'
     end
   end
 
+  def defendant
+    @defendant ||= @defendant_search.find(@defendant_uuid).first
+  end
+
   private
 
+  def set_defendant_if_required
+    defendant
+  end
+
+  def authorize_defendant_search
```

Comment:
[ditto](https://github.com/ministryofjustice/laa-court-data-ui/pull/231/files#r464481290)

Reply:
👍 amended to load_and_authorize_defendant_search